cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/JoinResult.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.api;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * Join Row which has all the records to be combined together
+ * @param <INPUT_RECORD>
+ */
+public class JoinResult<INPUT_RECORD> {
```

Comment:
this should just be an interface with the actual implementation separate from the api. Otherwise we have stuff like guava's ImmutableList getting forced into the api.